src/DetailsView/components/switcher.tsx

```diff
@@ -18,17 +18,26 @@ export interface SwitcherProps {
     pivotKey: DetailsViewPivotType;
 }
 
-export const Switcher = NamedSFC<SwitcherProps>('Switcher', props => {
-    const onRenderOption = (option: IDropdownOption): JSX.Element => {
+export interface SwitcherState {
+    selectedKey: DetailsViewPivotType;
+}
+
+export class Switcher extends React.Component<SwitcherProps, SwitcherState> {
+    constructor(props: SwitcherProps) {
+        super(props);
+        this.state = { selectedKey: props.pivotKey };
+    }
+
+    private onRenderOption(option: IDropdownOption): JSX.Element {
         return (
             <div className="switcher-dropdown-option" aria-hidden="true">
                 {option.data && option.data.icon && <Icon iconName={option.data.icon} />}
                 <span>{option.text}</span>
             </div>
         );
-    };
+    }
 
-    const onRenderTitle = (options: IDropdownOption[]): JSX.Element => {
+    private onRenderTitle(options: IDropdownOption[]): JSX.Element {
```

Comment:
Why are you changing the style of the functions here??? using the previous style let you avoid the use of `@autobind`

Reply:
Is it possible to create a arrow function in our class implementations? I think he changed the `Switcher` from a function to class and thats why he changed the arrow function to a private function as well.

Reply:
@Shobhit1, it is possible.